### PR TITLE
Enabled Ninja Generator in Windows CI

### PR DESCRIPTION
**Pull request by anitagov**
- Enabled Ninja Generator in Windows CI PS script and Jenkins stages.
- Removed NMake Makefiles Generator as Ninja replaces this.
- Renamed the 4 configurations to reflect what they are doing.
      - Win 2016 Debug renamed to Win 2016 Debug Cross Compile
      - Win 2016 Release renamed to Win 2016 Release Cross Compile
      - Win 2016 Debug Cross Platform renamed to Win 2016 Debug Linux-Elf-Build
      - Win 2016 Release Cross Platform renamed to Win 2016 Release Linux-Elf-Build
- Optimized 2 of the Windows stages to not run ctest on Linux after building ELF enclaves. 
- Removed the -OE_SIMULATION flag being passed incorrectly to cmake before Linux elf builds in Windows configurations.

### CODE REVIEW

**Comment by anitagov** [comment]
bors try

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -198,22 +196,22 @@ def win2016ReleaseCrossPlatform() {
...
 }
 
-def win2016Debug() {
+def win2016DebugCrossCompile() {
     stage('Windows Debug') {
         node('SGXFLC-Windows') {
             cleanWs()
             checkout scm
-            PowerShellWrapper('./scripts/test-build-config.ps1 -add_windows_enclave_tests -linux_bin_dir $ENV:WORKSPACE/linuxbin/tests -build_type Debug -build_enclaves')
+            PowerShellWrapper('./scripts/test-build-config.ps1 -add_windows_enclave_tests -build_type Debug -build_enclaves')
```
[the remark this answers is not in the document]
Comment (anitagov) [reply .jenkins/Jenkinsfile]: Good catch. Since fixed

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
@anitagov 

The last CI run reported [the following error](https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/Bors/detail/trying/665/pipeline/38):
```
-- The C compiler identification is Clang 7.0.1
CMake Error at C:/Program Files/CMake/share/cmake-3.13/Modules/CMakeDetermineCompilerId.cmake:802 (message):

  The Clang compiler tool

    "C:/Program Files/LLVM/bin/clang-7.exe"

  targets the MSVC ABI but has a GNU-like command-line interface.  This is
  not supported.  Use 'clang-cl' instead, e.g.  by setting 'CC=clang-cl' in
  the environment.

Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.13/Modules/CMakeDetermineCCompiler.cmake:113 (CMAKE_DIAGNOSE_UNSUPPORTED_CLANG)
  CMakeLists.txt:28 (project)


CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_ASM_MASM_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "D:/Jenkins/workspace/Bors_trying/build/X64-Debug/CMakeFiles/CMakeOutput.log".

cmake failed
```

I took a quick look over the updated `test-build-config.ps1` and I applied the following [diff](https://www.diffchecker.com/FiBLtNXX). After this, the build fails due to some warning treated as errors. 

[This](https://pastebin.com/raw/0jmuNxn2) is the full log of a manual run of `test-build-config.ps1` with my changes applied.

You may notice the following to my `test-build-config.ps1` script changes:

* I removed the hard-coded PATH append:
  ```
  $env:PATH += ";C:\Program Files\LLVM\bin"
  ```
  This is not only a quick hack, but it's not even needed. I double checked both our CI servers, and the LLVM is already present in the system PATH. I'd recommend that you remove it in this PR since it doesn't do anything.

* I set `CC` and `CXX` env variables to get pass the initial CI build error:
  ```
  $env:CC="clang-cl"
  $env:CXX="clang-cl"
  ```
  If you think this should be set by the script, please feel free to add them to the script.

Regards,
Ionut

**Comment by anitagov** [comment]
Thanks @ionutbalutoiu and @anakrish - I was updating the issue #1283 with my observations.
1) The Ninja Release built correctly and reported 2 tests as Failures (platform error). This is expected as we have never run Ninja in Release mode (reason we need Ninja in CI). I believe I saw these failures on my system too when I ran in Release mode.
2) Debug Mode - Why are we seeing build errors only on this platform but not on the other platform?
@anakrish - Is there a manual way to fire the Native X64 Command prompt? I am continuing to use the following to fire the Command window (same path is used for Developer Command Window and Native X64 Command prompt).
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvars64.bat

@ionutbalutoiu - Yes, we should not be adding the LLVM bin path in the script. The platform should be configured to have this in the path. Will try with your changes to set cc and ccx. Thanks.

**Comment by anitagov** [comment]
When you run cmake from the x64 Native Tools Command Prompt for VS 201 from build\X64-Debug directory, we should see the following:
-- The C compiler identification is MSVC 19.16.27026.1
-- The CXX compiler identification is MSVC 19.16.27026.1
-- The ASM_MASM compiler identification is MSVC

@ionutbalutoiu - Please verify if this works on the failing platform. If not, try rebooting the system. Thanks.

**Comment by anakrish** [comment]
@anitagov 

> This is expected as we have never run Ninja in Release mode

The Windows workflow based on Ninja has been tested on both Debug and Release builds.
There ought not to be any Release build specific errors on CI machines due to Ninja.

**Comment by anakrish** [comment]
@anitagov 

> 2\. Is there a manual way to fire the Native X64 Command prompt?

In interactive workflow (as opposed to scripted CI), the command prompt is launched from the Visual Studio start-menu folder. We want "x64 Native Prompt"; and that is what the CI workflow needs to use as well for consistency. Note, VS start folder has many prompts, only some work for us.

**Comment by anitagov** [comment]
ecall_ocall and filetest fail in Release mode. I did another sandbox run and got the Debug ninja to run on the correctly configured platform. ecall_ocall failed there too with platform_error. The size of the enclave may need to be reduced further or we really need to use the Native X64 Command Prompt. Figuring a way to fire this. The path appears to be the same as the Developer Command Prompt i.e. C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvars64.bat

**Comment by anitagov** [comment]
@anakrish - I verified that the Native X64 Command prompt and Developer Command Prompt are identical but for the title. Both execute the same command as I mentioned earlier i.e. 
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvars64.bat.

One of the CI platforms is not configured correctly and that still needs to be fixed.

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
Thanks to John for rebooting the Windows CI platforms but that did not help in fixing the botched Visual Studio Tools installation on one of the platforms. Found out that "where cl" returned the correct path in both the systems.
Hard-coded the CC and CXX environment variables to point to cl and get CI to work. Will trouble-shoot the faulty platform tomorrow.

**Comment by anitagov** [comment]
bors try

**Pull request merged**